src/newgrf_roadstop.cpp

```diff
@@ -171,6 +171,17 @@ uint32 RoadStopScopeResolver::GetVariable(byte variable, uint32 parameter, bool
 			return ssl.grfid;
 		}
 
+		/* 16 bit Station ID of nearby tiles */
+		case 0x6B: { 
+			TileIndex nearby_tile = GetNearbyTile(parameter, this->tile);
+
+			if (!HasStationTileRail(nearby_tile)) return 0xFFFF;
+			if (!IsCustomStationSpecIndex(nearby_tile)) return 0;
+
+			const StationSpecList ssl = BaseStation::GetByTile(nearby_tile)->speclist[GetCustomStationSpecIndex(nearby_tile)];
```

Comment:
This should use IsRoadStopTile, IsCustomRoadStopSpecIndex, RoadStopSpecList, roadstop_speclist and GetCustomRoadStopSpecIndex instead of the rail station functions/types.